packages/headless-react/src/ssr-commerce/commerce-engine.test.tsx

```diff
@@ -37,8 +37,15 @@ describe('Headless react SSR utils', () => {
     errorSpy = vi.spyOn(console, 'error');
   });
 
-  afterEach(() => {
+  afterEach(async () => {
     errorSpy.mockClear();
+
+    await act(async () => {
+      await new Promise((resolve) => setTimeout(resolve, 0));
+    });
```

Comment:
We could extract this into a util function (e.g., `flushAsyncUpdates`) rather than repeating it.